internal/pgbackrest/config.go

```diff
@@ -465,6 +466,20 @@ func serverConfig(cluster *v1beta1.PostgresCluster) iniSectionSet {
 	// - https://releases.k8s.io/v1.23.0/pkg/kubelet/kubelet_pods.go#L345
 	global.Set("tls-server-address", "0.0.0.0")
 
+	// NOTE (dsessler7): As pointed out by Chris above, there is an issue in
+	// pgBackRest (#1841), where using a wildcard address to bind all addresses
+	// does not work in certain IPv6 environments. Until this is fixed, we are
+	// going to workaround the issue by allowing the user to add an annotation to
+	// enable IPv6. We will check for that annotation here and override the
+	// "tls-server-address" setting accordingly.
+	annotations := cluster.GetAnnotations()
+	if annotations != nil {
+		if ipVersion, exists := annotations[naming.PGBackRestIPVersion]; exists &&
+			strings.ToLower(ipVersion) == "ipv6" {
```

Comment:
👍🏻 The docs all say (capital) `IPv6` and I'm fully on board with being case-insensitive here.

⛏️ If we're trying to do a case-insensitive comparison, we should go ahead and use the [function for that](https://pkg.go.dev/strings#EqualFold) `strings.EqualFold(ipVersion, "ipv6")`.

Reply:
I wasn't aware of that function, but good to know.